pkg/controller/servicebindingrequest/annotations/annotations.go

```diff
@@ -0,0 +1,88 @@
+package annotations
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+// bindingType encodes the medium the binding should deliver the configuration value.
+type bindingType string
+
+const (
+	// BindingTypeVolumeMount indicates the binding should happen through a volume mount.
+	BindingTypeVolumeMount bindingType = "volumemount"
+	// BindingTypeEnvVar indicates the binding should happen through environment variables.
+	BindingTypeEnvVar bindingType = "env"
+)
+
+// supportedBindingTypes contains all currently supported binding types.
+var supportedBindingTypes = map[bindingType]bool{
+	BindingTypeVolumeMount: true,
+	BindingTypeEnvVar:      true,
+}
+
+// dataPath is the path ConfigMap and Secret resources use to scope their data.
+//
+// note: it is currently used to provide a pointer to the "data" string, which is the location
+// ConfigMap and Secret resources keep user data.
+var dataPath = "data"
+
+// Result contains data that has been collected by an annotation handler.
+type Result struct {
+	// Object is the data that has been collected by the Handler; it can be found in the location
+	// indicated by the Path field.
+	Object map[string]interface{}
+	// Type indicates where the Object field should be injected in the application; can be either
+	// "env" or "volumemount".
+	Type bindingType
+	// Path is the nested location the collected data can be found in the Object field.
+	Path string
+}
+
+// Handler should be implemented by types that want to offer a mechanism to provide binding data to
+// the system.
+type Handler interface {
+	// Handle returns binding data.
+	Handle() (Result, error)
+}
+
+type ErrHandlerNotFound string
+
+func (e ErrHandlerNotFound) Error() string {
+	return fmt.Sprintf("could not find handler for annotation value %q", string(e))
+}
+
+func IsErrHandlerNotFound(err error) bool {
+	_, ok := err.(ErrHandlerNotFound)
+	return ok
+}
+
+// BuildHandler attempts to create an annotation handler for the given annotationKey and
+// annotationValue. kubeClient is required by some annotation handlers, and an error is returned in
+// the case it is required by an annotation handler but is not defined.
+func BuildHandler(
```

Comment:
We don't have a test for this particular function. Is it intentional?

Reply:
Didn't add it because it usage has been covered by the other existing tests.

Reply:
Hm, but code coverage reports 0% for this file:

https://codecov.io/gh/redhat-developer/service-binding-operator/pull/407/diff?src=pr&el=tree#diff-cGtnL2NvbnRyb2xsZXIvc2VydmljZWJpbmRpbmdyZXF1ZXN0L2Fubm90YXRpb25zL2Fubm90YXRpb25zLmdv

Reply:
Ideally, every function ( "unit" ) should have a test.

Reply:
Hm, indeed strange. The caller is being covered in the tests though:

![image](https://user-images.githubusercontent.com/94526/81739614-f1746400-949b-11ea-86c3-c7b536c5e9e4.png)